xdsl/utils/dialect_stub.py

```diff
@@ -0,0 +1,238 @@
+from dataclasses import dataclass, field
+from typing import Annotated, Any
+
+from xdsl.dialects.builtin import ArrayAttr, ArrayOfConstraint
+from xdsl.interpreter import Successor
+from xdsl.ir import (
+    Attribute,
+    Dialect,
+    OpResult,
+    OptSuccessor,
+    ParametrizedAttribute,
+    Region,
+    VarSuccessor,
+)
+from xdsl.irdl import (
+    AllOf,
+    AnyAttr,
+    AnyOf,
+    AttrConstraint,
+    AttributeDef,
+    BaseAttr,
+    EqAttrConstraint,
+    IRDLOperation,
+    Operand,
+    OperandDef,
+    OptAttributeDef,
+    OptOperand,
+    OptOperandDef,
+    OptOpResult,
+    OptPropertyDef,
+    OptRegion,
+    OptRegionDef,
+    OptResultDef,
+    OptSuccessorDef,
+    ParamAttrConstraint,
+    PropertyDef,
+    RangeConstraint,
+    RegionDef,
+    ResultDef,
+    SingleOf,
+    SuccessorDef,
+    VarOperand,
+    VarOperandDef,
+    VarOpResult,
+    VarRegion,
+    VarRegionDef,
+    VarResultDef,
+    VarSuccessorDef,
+)
+
+
+@dataclass
+class DialectStub:
+    dialect: Dialect
+    dependencies: dict[Any, set[tuple[Any, str]]] = field(
+        init=False, default_factory=dict
+    )
+
+    def _import(self, object: Any, alias: str = ""):
+        module = object.__module__
+        if module == "builtins":
+            return
+        if module.startswith("xdsl.ir."):
+            module = "xdsl.ir"
+        if module.startswith("xdsl.irdl."):
+            module = "xdsl.irdl"
```

Comment:
this seems a little hacky, don't we want to make this work anywhere, including in dependencies of xDSL?

Reply:
What do you mean by make this work anywhere?

This simply removes the banned nested modules to make Pyright happy about the output stubs, it, don't import from `xdsl.ir.core` but just from `xdsl.ir`. Then, be it one of those modules or any other one, it will be imported, including external dependencies. Note that, this is intended to generate stubs for parsed irdl dialect, so I'm not sure when irdl will reach a stage able to map to external Python dependencies?

The first line is about not importing builtin types like `int` or `str` - not sure which one triggered it now :thinking:

Reply:
Ah, I see. I think I need to check out the branch and play with it a little bit for more insightful comments than "this looks weird" :)

Reply:
I guess I can offer comments too 😉